src/sentry/net/socket.py

```diff
@@ -100,8 +101,15 @@ def is_safe_hostname(hostname: str | None) -> bool:
 
 # Modifed version of urllib3.util.connection.create_connection.
 def safe_create_connection(
-    address, timeout=socket._GLOBAL_DEFAULT_TIMEOUT, source_address=None, socket_options=None
+    address,
+    timeout=socket._GLOBAL_DEFAULT_TIMEOUT,
+    source_address=None,
+    socket_options=None,
+    is_ipaddress_permitted: Callable[[str], bool] | None = is_ipaddress_allowed,
```

Comment:
```suggestion
    is_ipaddress_permitted: Callable[[str], bool] | None = None,
```

Should this be `None` as you also default to `is_ipaddress_allowed` on 111.